streams/src/test/java/org/apache/kafka/test/MockInternalProcessorContextTest.java

```diff
@@ -0,0 +1,287 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.test;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.header.internals.RecordHeader;
+import org.apache.kafka.common.header.internals.RecordHeaders;
+import org.apache.kafka.common.metrics.Metrics;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.StreamsConfig;
+import org.apache.kafka.streams.processor.AbstractNotifyingRestoreCallback;
+import org.apache.kafka.streams.processor.StateRestoreCallback;
+import org.apache.kafka.streams.processor.StateStore;
+import org.apache.kafka.streams.processor.internals.CompositeRestoreListener;
+import org.apache.kafka.streams.processor.internals.InternalProcessorContext;
+import org.apache.kafka.streams.processor.internals.ProcessorRecordContext;
+import org.apache.kafka.streams.processor.internals.RecordBatchingStateRestoreCallback;
+import org.apache.kafka.streams.processor.internals.metrics.StreamsMetricsImpl.Version;
+import org.easymock.Capture;
+import org.easymock.EasyMock;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+
+import static org.apache.kafka.test.MockInternalProcessorContext.DEFAULT_TASK_ID;
+
+public class MockInternalProcessorContextTest {
+
+    @Test
+    public void shouldReturnDefaults() {
+        final MockInternalProcessorContext context = new MockInternalProcessorContext();
+
+        verifyDefaultMetricsVersion(context);
+        verifyDefaultRecordCollector(context);
+        verifyDefaultTaskId(context);
+        verifyDefaultTopic(context);
+        verifyDefaultPartition(context);
+        verifyDefaultTimestamp(context);
+        verifyDefaultOffset(context);
+        verifyDefaultHeaders(context);
+        verifyDefaultProcessorNodeName(context);
```

Comment:
req: Please do not add a method for each default value. IMO using `assertThat()` as I did in the following is readable enough.
  
```suggestion
        assertThat(context.metrics().version(), is(Version.LATEST));
        assertThat(context.recordCollector(), notNullValue());
        assertThat(context.taskId(), is(MockInternalProcessorContext.DEFAULT_TASK_ID));
        assertThat(context.recordContext().topic(), is(MockInternalProcessorContext.DEFAULT_TOPIC));
        assertThat(context.recordContext().partition(), is(MockInternalProcessorContext.DEFAULT_PARTITION));
        assertThat(context.recordContext().timestamp(), is(MockInternalProcessorContext.DEFAULT_TIMESTAMP));
        assertThat(context.recordContext().offset(), is(MockInternalProcessorContext.DEFAULT_OFFSET));
        assertThat(context.recordContext().headers(), is(MockInternalProcessorContext.DEFAULT_HEADERS));
        assertThat(context.currentNode().name(), is(MockInternalProcessorContext.DEFAULT_PROCESSOR_NODE_NAME));
```